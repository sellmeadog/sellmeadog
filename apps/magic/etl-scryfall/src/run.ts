import { make_scryfall_data_source } from '@sellmeadog/data-source/scryfall';
import { transform_scryfall_set } from '@sellmeadog/magic-transformation';
import { concat, lastValueFrom, map } from 'rxjs';

const scryfall = make_scryfall_data_source();

export async function run() {
  const sets_etl$ = scryfall
    .sets()
    .pipe(map((sets) => sets.map(transform_scryfall_set)));

  return await lastValueFrom(concat(sets_etl$));
}
