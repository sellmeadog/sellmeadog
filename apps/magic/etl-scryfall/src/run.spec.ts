import { makeMockedFunction } from '@sellmeadog/extensions/testing';
import { createReadStream } from 'fs';
import fetch, { Response } from 'node-fetch';
import { resolve } from 'path';
import { run } from './run';

describe('main', () => {
  let fetch_: jest.MockedFunction<typeof fetch>;

  beforeEach(() => {
    jest.resetAllMocks();
    fetch_ = makeMockedFunction(fetch);
  });

  it('should run', async () => {
    const rs = createReadStream(resolve(__dirname, '__json__', 'sets.json'));
    fetch_.mockResolvedValue(new Response(rs));

    expect(await run()).toMatchSnapshot();
  });
});
