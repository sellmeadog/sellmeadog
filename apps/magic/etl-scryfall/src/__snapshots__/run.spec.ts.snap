// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should run 1`] = `
Array [
  Object {
    "_id": "bc94aba1-7376-4e02-a12d-3a2efb66ab0f",
    "block": "Core Set",
    "block_code": "lea",
    "card_count": Long {
      "high": 0,
      "low": 397,
      "unsigned": false,
    },
    "code": "m21",
    "digital": false,
    "icon_svg_uri": "https://c2.scryfall.com/file/scryfall-symbols/sets/m21.svg?1646024400",
    "name": "Core Set 2021",
    "parent_set_code": null,
    "printed_size": Long {
      "high": 0,
      "low": 397,
      "unsigned": false,
    },
    "released_at": "2020-07-03",
    "set_type": "core",
    "tcgplayer_id": Long {
      "high": 0,
      "low": 2653,
      "unsigned": false,
    },
  },
  Object {
    "_id": "4887d21f-71e7-4d7a-a079-e9521fd7e6d7",
    "block": "Non-Block Expansion",
    "block_code": "ss3",
    "card_count": Long {
      "high": 0,
      "low": 8,
      "unsigned": false,
    },
    "code": "ss3",
    "digital": false,
    "icon_svg_uri": "https://c2.scryfall.com/file/scryfall-symbols/sets/ss3.svg?1646024400",
    "name": "Signature Spellbook: Chandra",
    "parent_set_code": null,
    "printed_size": Long {
      "high": 0,
      "low": 8,
      "unsigned": false,
    },
    "released_at": "2020-06-26",
    "set_type": "spellbook",
    "tcgplayer_id": Long {
      "high": 0,
      "low": 2633,
      "unsigned": false,
    },
  },
]
`;
